### DISCUSSION TITLE
I am unable to create the a.exe file for my tut1.cpp file! (I am new to programming!)

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/vscode-cpptools#9766] [Q&A] [answered]
### **I am unable to create the a.exe file for my tut1.cpp file! (I am new to programming!)**
this is what ive been showed.
![image](https://user-images.githubusercontent.com/111668690/185771927-1926741e-09c1-430f-8ac9-db424f93dad6.png)
**I have installed the MinGw compiler and added it to my path already.**
![image](https://user-images.githubusercontent.com/111668690/185771968-77a03d92-34ca-49de-adc1-13f412af80c7.png)
**and now still ive been shown with this error and im confused. someone pleasee aid me.!!**

**This is the response given to me by the terminal.**

**```
Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.
Install the latest PowerShell for new features and improvements! https://aka.ms/PSWindows
PS C:\myprogram\c++ course> g++ .\tut1.cpp
c:/mingw/bin/../lib/gcc/mingw32/9.2.0/../../../../mingw32/bin/ld.exe: c:/mingw/bin/../lib/gcc/mingw32/9.2.0/../../../libmingw32.a(main.o):(.text.startup+0xc0): undefined reference to `WinMain@16'
collect2.exe: error: ld returned 1 exit status
PS C:\myprogram\c++ course> g++ .\tut1.cpp
c:/mingw/bin/../lib/gcc/mingw32/9.2.0/../../../../mingw32/bin/ld.exe: c:/mingw/bin/../lib/gcc/mingw32/9.2.0/../../../libmingw32.a(main.o):(.text.startup+0xc0): undefined reference to `WinMain@16'
collect2.exe: error: ld returned 1 exit status
PS C:\myprogram\c++ course>
```**

**Comment by sean-mcmanus** [comment, marked as answer]
Have you not saved your file? It's complaining about not finding a main method and it reads the file from disk.

Also, you may need a tasks.json. You can try using the Debug C/C++ option in the top right of the file (it seems like you may be using g++ directly from the command line).
